### PR TITLE
User feature

### PR DESCRIPTION
**Pull request by GitChrisCode**
This PR provides the functionality to register a new user with a name and password via a simple frontend.

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/repository/RecipeRepository.java
Comment (MJaschewski) [inline backend/src/main/java/de/neuefische/backend/repository/RecipeRepository.java:1]: Either change to make it useful or remove.

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class UserTasteBuddiesControllerTest {
+
+
+    private UserTasteBuddiesController controller;
```
Comment (MJaschewski) [inline backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: Remove this unused controller

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    MongoUserTasteBuddiesRepo userRepo = mock(MongoUserTasteBuddiesRepo.class);
+    PasswordEncoder passwordEncoder = mock(PasswordEncoder.class);
+    UUIDService uuidService = mock(UUIDService.class);
+    UserTasteBuddiesDetailsService service = new UserTasteBuddiesDetailsService(userRepo, passwordEncoder, uuidService);
+    @Autowired
```
Comment (MJaschewski) [inline backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java:17]: Put Autowired as first line after intialising class

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                        .param("userName", userName)
+                        .param("userPassword", userPassword)
+                )
+                .andExpect(status().isCreated())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.userName").value(userName));
+
+    }
+
+    @Test
+    void testRegisterUserTasteBuddiesV2() {
```
Comment (MJaschewski) [inline backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: Naming! This does not tell me what this does

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        UserTasteBuddiesDetailsService mockService = mock(UserTasteBuddiesDetailsService.class);
+
+        UserTasteBuddies expectedUser = new UserTasteBuddies();
+        expectedUser.setUserName("testUser");
+        expectedUser.setUserPassword("testPassword");
+        when(mockService.registerUserTasteBuddies(anyString(), anyString())).thenReturn(expectedUser);
```
Comment (MJaschewski) [inline backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: You don't use mock in integration tests.

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+        assertEquals(expectedUser, resultUser);
+    }
+
+
+    @Test
+    void testLoadUserByName() {
+        String username = "noUser";
+
+        when(userRepo.findUserByUserName(username)).thenReturn(Optional.empty());
```
Comment (MJaschewski) [inline backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: dont use mocks in integration tests.

File: frontend/package.json [hunk 6]
```diff
@@ -41,5 +42,8 @@
       "last 1 safari version"
     ]
   },
-  "proxy": "http://localhost:8080"
+  "proxy": "http://localhost:8080",
+  "devDependencies": {
```
Comment (MJaschewski) [inline frontend/package.json]: Dont use devDependencies

**Review by MJaschewski — CHANGES_REQUESTED** [review]
Looks nice, but there are some changes needed.

File: backend/src/main/java/de/neuefische/backend/repository/RecipeRepository.java
Comment (GitChrisCode) [reply backend/src/main/java/de/neuefische/backend/repository/RecipeRepository.java:1]: Will be soon used ...

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java (hunk 3 above)
Comment (GitChrisCode) [reply backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: changed

File: frontend/package.json (hunk 6 above)
Comment (GitChrisCode) [reply frontend/package.json]: changed

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java (hunk 4 above)
Comment (GitChrisCode) [reply backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: moved content in service test

File: backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java (hunk 5 above)
Comment (GitChrisCode) [reply backend/src/test/java/de/neuefische/backend/controller/UserTasteBuddiesControllerTest.java]: moved content in service test

**Comment by GitChrisCode** [comment]
issues resolved

**Review by MJaschewski — APPROVED** [review]

**Pull request merged**